src/App.tsx

```diff
@@ -15,48 +15,113 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  BY_NONE = 0,
+  BY_ALPHABET,
+  BY_LENGTH,
+}
+
+type SortQuery = {
+  sortBy: SortType;
+  isSortReversed: boolean;
+};
+
+function getPreparedGoods(
+  goods: string[],
+  { sortBy, isSortReversed }: SortQuery,
+) {
+  const preparedGoods = [...goods].sort((a, b) => {
+    switch (sortBy) {
+      case SortType.BY_ALPHABET:
+        return a.localeCompare(b);
+
+      case SortType.BY_LENGTH:
+        return (a.length - b.length);
+
+      default:
+        return 0;
+    }
+  });
+
+  return isSortReversed ? preparedGoods.reverse() : preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortBy, setSortBy] = useState(SortType.BY_NONE);
+  const [isSortReversed, setIsSortReversed] = useState(false);
+  const handleSortByAlphabet = () => {
+    setSortBy(sortBy === SortType.BY_ALPHABET
+      ? SortType.BY_NONE
+      : SortType.BY_ALPHABET);
+  };
+
+  const handleSortByLength = () => {
+    setSortBy(sortBy === SortType.BY_LENGTH
+      ? SortType.BY_NONE
+      : SortType.BY_LENGTH);
+  };
+
+  const handleSortReversed = () => {
+    setIsSortReversed((prev = isSortReversed) => {
+      return !prev;
+    });
+  };
```

Comment:
```suggestion
    setIsSortReversed(prev =>  !prev);
  };
```
why do you need to set default value for previous State?